warrior/Framework/ClassUtils/gnmi_utils_class.py

```diff
@@ -0,0 +1,339 @@
+        """
+        status = False
+        if external_system == None:
+            child = pexpect.spawn(cmd_string)
+        else:
+            session_id = data_Utils.get_session_id(external_system, external_system_session)
+            child = data_Utils.get_object_from_datarepository(session_id)
+            child.sendline(cmd_string)
+        while 1:
+            try:
+                u_index = child.expect(['password', 'Permission denied', pexpect.EOF,
+                                        'want to continue connecting', ".*100%.*(%|#|r'\$')",
+                                        'Connection reset by peer', pexpect.TIMEOUT], timeout=20)
+                if u_index == 0:
+                    child.sendline(passwd)
+                    testcase_Utils.pNote(child.before+child.match.group(0))
+                if u_index == 1:
+                    status = False
+                    testcase_Utils.pNote(child.before+child.match.group(0)+child.after)
+                    break
+                if u_index == 2:
+                    testcase_Utils.pNote(child.before)
+                    if "100%" in child.before:
+                        status = True
+                    break
+                if u_index == 4:
+                    testcase_Utils.pNote(child.before+child.match.group(0))
+                    if "100%" in child.before+child.match.group(0):
+                        status = True
+                    break
+                if u_index == 3:
+                    testcase_Utils.pNote(child.before+child.match.group(0)+child.after)
+                    child.sendline('yes')
+                if u_index == 5:
+                    testcase_Utils.pNote(child.before+child.match.group(0))
+                    status = False
+                    break
+            except:
+                testcase_Utils.pNote("File Copy Failed", "error")
+                status = False
+                break
+        if status:
+            testcase_Utils.pNote("Client certificate copied Successfully.")
+        else:
+            testcase_Utils.pNote("Client certificate copy Failed.", "error")
+        return status
+
+    def execuate(self, cmd_string, uname, passwd, external_system=None,
+                 external_system_session=None, stop_after=None, gnmi_obj=None, script="No"):
+        """
+        Execute gNMI command using gNMI binary
+        :param cmd_string:
+        :param uname:
+        :param passwd:
+        :param external_system:
+        :param external_system_session:
+        :param stop_after:
+        :param gnmi_obj:
+        :param script:
+        :return:
+        """
+        status = False
+        result = None
+        excute = False
+        child = None
+        if external_system == None:
+            if gnmi_obj == None:
+                child = pexpect.spawn(cmd_string)
+                child.maxread = 50000
+            else:
+                child = gnmi_obj
+                child = pexpect.spawn(cmd_string)

```

Comment:
what does lines 115, 116 do. 
child is intialized as gnmi_obj in line 115
child is again set to pexpect.spawn(cmd_string) in line 116 

what is the effect of line 115 when it immediately overwritten in the next line. 
and going by this code i dont see any difference when gnmi_obj = None and when gnmi_obj is not None.

Are these lines the reason why the user is seeing a failure, can you please check and address the issue in this area.